services/app-api/src/postgres/contractAndRates/parseContractWithHistory.ts

```diff
@@ -194,29 +194,26 @@ function contractWithHistoryToDomainModel(
         //     }
         // }
 
-        // Basically the same as above, except we do not create new contract revisions for rate changes.
+        // Basically the same as above, except we do not create new contract revisions for rate changes and non-draft
+        //  contract revisions do not inlcude rate revisions that where submitted after the contract revision was submitted.
         for (const rateRev of contractRev.rateRevisions) {
```

Comment:
I think this could be simpler. 

What we're saying right now is that we are only going to return the rates that existed when the contract was submitted, and ignore all rates that came afterwards. 

So I think this whole for loop becomes

```
initialEntry.rateRevisons = contractRev.rateRevisions.filter((rr) => rr.rateRevision.submitInfo.updatedAt <=
                contractRev.submitInfo.updatedAt)
```

Reply:
and I think we should be making the same change in parseRateWithHistory b/c that's what turns into the rate dashboard.

Reply:
```ts
initialEntry.rateRevisons = contractRev.rateRevisions.filter((rr) => rr.rateRevision.submitInfo.updatedAt <=
                contractRev.submitInfo.updatedAt)
```

This wouldn't remove the rates that had been removed. Also, using the array method, we would have to check the `submitInfo` on both revisions for `undefined` again.

I did simplify the loop a bit more.

Reply:
> and I think we should be making the same change in parseRateWithHistory b/c that's what turns into the rate dashboard.

This one is slightly different from contract revisions since the rates are submitted before the contract using the `submitHealthPlanPackage` resolver.